components/eventing-controller/testing/test_helpers.go

```diff
@@ -304,44 +330,61 @@ func WithFilter(eventSource, eventType string) SubscriptionOpt {
 	}
 }
 
-func WithEmptyFilter(subscription *eventingv1alpha1.Subscription) {
-	subscription.Spec.Filter = &eventingv1alpha1.BEBFilters{
-		Filters: []*eventingv1alpha1.BEBFilter{},
-	}
+// WithNotCleanFilter initializes subscription filter with a not clean event-type
+// A not clean event-type means it contains none-alphanumeric characters
+func WithNotCleanFilter() SubscriptionOpt {
+	return WithFilter(EventSource, OrderCreatedEventTypeNotClean)
 }
 
-func WithEventTypeFilter(s *eventingv1alpha1.Subscription) {
-	s.Spec.Filter = &eventingv1alpha1.BEBFilters{
-		Filters: []*eventingv1alpha1.BEBFilter{
-			{
-				EventSource: &eventingv1alpha1.Filter{
-					Type:     "exact",
-					Property: "source",
-					Value:    EventSource,
-				},
-				EventType: &eventingv1alpha1.Filter{
-					Type:     "exact",
-					Property: "type",
-					Value:    OrderCreatedEventType,
-				},
-			},
-		},
+// WithEmptyFilter is a SubscriptionOpt for creating a subscription with an empty event type filter.
+//  Note that this is different from setting Filter to nil.
+func WithEmptyFilter() SubscriptionOpt {
+	return func(subscription *eventingv1alpha1.Subscription) {
+		subscription.Spec.Filter = &eventingv1alpha1.BEBFilters{
+			Filters: []*eventingv1alpha1.BEBFilter{},
+		}
 	}
 }
 
-// WithSinkURL sets the sinkURL in a new subscription
-func WithSinkURL(sinkURL string) SubscriptionOpt {
-	return func(s *eventingv1alpha1.Subscription) {
-		s.Spec.Sink = sinkURL
-	}
+func WithOrderCreatedFilter() SubscriptionOpt {
```

Comment:
Good question:

Reply:
`WithFilterOrderCreated` sounds like we are filtering some `OrderCreated` items. Putting `Filter` at the end of the words makes --in my option-- clear, that this handles some kind of a `Filter` object.

Reply:
I personally like `WithFilterOrderCreated` more because it enables easier auto completion, but I agree that `WithOrderCreatedFilter` sounds better. No need to change for me.